### PR TITLE
Integration test for Istio DNS certificates generated by Chiron

### PR DESCRIPTION
**Pull request by lei-tang**
Please provide a description for what this PR is for: https://github.com/istio/istio/issues/17061.

Design documents:
[1]https://docs.google.com/document/d/1VHyBre8943USOx_YEVtA18KfEoGTmLT1iBHESO5bzcA/edit#heading=h.qex63c29z2to
[2] https://docs.google.com/document/d/1v8BxI07u-mby5f5rCruwF7odSXgb9G8-C9W5hQtSIAg/edit#heading=h.xw1gqgyqs5b

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ X] Security
[X ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: tests/integration/security/chiron/main_test.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	inst istio.Instance
+	g    galley.Instance
+	p    pilot.Instance
+)
+
+func TestMain(m *testing.M) {
+	// Integration test for provisioning DNS certificates.
+	framework.NewSuite("dns_certificate_test", m).
+		Label(label.CustomSetup).
+		SetupOnEnv(environment.Kube, istio.Setup(&inst, setupConfig)).
```
Comment (howardjohn) [inline tests/integration/security/chiron/main_test.go]: why do you even need to install all of Istio for this? isn't it standalone?

Comment (lei-tang) [reply tests/integration/security/chiron/main_test.go]: Chiron is linked into Pilot.

**Review by incfly — COMMENTED** [review]
Can we avoid customized setup? Or making all the Helm changes you need in the shared test setup for the main test suite under `test/security/`?

We're moving them together to reduce the time, and should be vigilant to increase the customized Istio setup in new test suite and avoid them if possible.

File: tests/integration/security/chiron/main_test.go (hunk 1 above)
Comment (howardjohn) [reply tests/integration/security/chiron/main_test.go]: I misunderstood this, sgtm

File: tests/integration/security/chiron/main_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		}).
+		Run()
+}
+
+func setupConfig(cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+
+	// Helm values from install/kubernetes/helm/istio/values-istio-dns-cert.yaml
```
Comment (howardjohn) [inline tests/integration/security/chiron/main_test.go]: this folder is shipped to users, should we just use the `cfg.Values` to configure it inline?

**Comment by lei-tang** [comment]
> Can we avoid customized setup? Or making all the Helm changes you need in the shared test setup for the main test suite under `test/security/`?
> 
> We're moving them together to reduce the time, and should be vigilant to increase the customized Istio setup in new test suite and avoid them if possible.

Done: following the convention of other test helm values, moved it to install/kubernetes/helm/istio/test-values/values-istio-dns-cert.yaml.

File: tests/integration/security/chiron/main_test.go (hunk 2 above)
Comment (lei-tang) [reply tests/integration/security/chiron/main_test.go]: Done: following the convention of other test helm values, moved it to install/kubernetes/helm/istio/test-values/values-istio-dns-cert.yaml.

File: install/kubernetes/helm/istio/test-values/values-istio-dns-cert.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+global:
```
Comment (incfly) [inline install/kubernetes/helm/istio/test-values/values-istio-dns-cert.yaml]: I was actually thinking to move these to `test-values/values-e2e.yaml`, and then delete `chiron/main_test.go`, move your  `dns_cert_test.go` to test/security.

@howardjohn I think it's fine to introduce these values into the shared values file, right?

Comment (lei-tang) [reply install/kubernetes/helm/istio/test-values/values-istio-dns-cert.yaml]: Added a TODO item to investigate whether this test can be moved to test/security. 1.4 deadline is approaching and the test placement can be optimized later.

Comment (lei-tang) [reply install/kubernetes/helm/istio/test-values/values-istio-dns-cert.yaml]: Had a discussion with @JimmyCYJ on merging the tests. Merging multiple tests in the same Istio deployment may interfere with each other and may cause the test results undeterministic. To make the tests run faster, the tests should run in parallel.

Comment (incfly) [reply install/kubernetes/helm/istio/test-values/values-istio-dns-cert.yaml]: @diemtvu thoughts?

**Review by howardjohn — APPROVED** [review]
Test changes look fine

**Review by myidpt — APPROVED** [review]

**Pull request merged**